codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectAggregator.java

```diff
@@ -174,8 +179,11 @@ protected boolean closeAfterContinueResponse(Object msg) {
 
     @Override
     protected boolean ignoreContentAfterContinueResponse(Object msg) {
-        return msg instanceof HttpResponse &&
-               ((HttpResponse) msg).status().code() == HttpResponseStatus.EXPECTATION_FAILED.code();
+        if (msg instanceof HttpResponse) {
+            final HttpResponse httpResponse = (HttpResponse) msg;
+            return httpResponse.status().code() >= 400 && httpResponse.status().code() < 500;
```

Comment:
consider using the `HttpStatusClass` instead (e.g. `((HttpResponse) msg).status().codeClass().equals(CLIENT_ERROR)`)